web/src/app/schedules/CalendarEventWrapper.js

```diff
@@ -190,7 +225,14 @@ export default class CalendarEventWrapper extends Component {
         }}
         title={this.renderInteractiveTooltip()}
       >
-        {children}
+        {React.cloneElement(children, {
+          onClick: () => {
+            children.props.onClick() // toggles selected
```

Comment:
children needs to be treated as opaque (don't assume anything about it)

To map items and set props use `React.Children.map`

It might be easier to use a container/wrapper and "catch" the propagated events instead though

Reply:
What I need to happen is trigger both the `onClick` given to the div by the calendar to change colors, and the `onClick` that we define to open the tooltip. I originally had a second div wrapping the child, but then the child div is never clicked on and only the top level `onClick` (to open the tooltip) would trigger. So we won't want to overwrite the `onClick` with a map, since we want both to trigger.

I'll do some research into event propagation from a child component. Maybe I can have a listener rather than another `onClick` inside of here.

Reply:
I found a good alternative that just uses a different click handler that doesn't conflict with the calendar events 🙂